### PR TITLE
Improve scan-all feature

### PR DESCRIPTION
**Pull request by caggle**
Basically to implement some/all/more of TODOs listed here: https://github.com/mozilla/vautomator-serverless/pull/40#issuecomment-501558039.

### CODE REVIEW

**Comment by caggle** [comment]
- [ ] Update documentation to reflect this feature.
- [ ] Consider implementing a state to beautify the email message coming from SNS.
- [ ] Consider implementing a state to fetch/download Tenable.io scan results in JSON. Currently the downloaded results do not include Tenable.io scan results.
- [ ] Make the state machine smarter, e.g. instead of running these tasks sequentially, all scans should originate from a port scan as a source of truth. For instance, only run SSH scan if an SSH port is open, only run TLS Observatory scan if a TLS service is offered etc. This is a longer term implementation.

**Comment by caggle** [comment]
This is ready for review.

The long term implementation mentioned above is rather a big piece, which would require its own PR.

This PR enables:
- A single endpoint which consumes a host and performs all types of supported scans on it,
- Waiting until most if not all scans to finish,
- Packaging the results, upload to S3 and generate a download link (signed S3 URL),
- Sending an email to (currently) 3 individuals with the download link and scan status for each type of scan (success/fail).

The resulting email looks like this:
image

File: docs/usage.rst [hunk 1]
```diff
@@ -62,6 +62,32 @@ There are 3 clients:
...
+Example
+++++++++
+.. parsed-literal::
+   curl -X POST 'https://y2ippncfd1.execute-api.us-west-2.amazonaws.com/dev/scan' \
+   -d '{"target": "www.mozilla.org"}' -H 'X-Api-Key: abcdefgh12345678'
+
+   {"executionArn":"<executionARN>:ScanAll:e9648493-9c01-11e9-85f4-874b479eba5f","startDate":1.561986763711E9}
+
+.. note:: This is an asynchronous endpoint. Behind the scenes, the host is processed 
+   by a state machine, which invokes a number of Lambda functions tp perform all scans
```
Comment (claudijd) [inline docs/usage.rst]: typo (tp)

File: lib/custom_exceptions.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class TenableScanRunningException(Exception):
+    pass
```
Comment (claudijd) [inline lib/custom_exceptions.py]: What is the purpose of these?

File: lib/tenableio_scan_handler.py [hunk 3]
```diff
@@ -7,22 +7,31 @@
 from lib.response import Response
 from lib.hosts import Hosts
 from lib.event import Event
+from scanners.tenable_io_scanner import TIOScanner
+# from lib.custom_exceptions import TenableScanCompleteException
```
Comment (claudijd) [inline lib/tenableio_scan_handler.py]: Unneeded?

File: lib/tenableio_scan_handler.py [hunk 4]
```diff
@@ -7,22 +7,31 @@
...
+from lib.s3_helper import send_to_s3
+
+S3_CLIENT = boto3.client('s3', region_name='us-west-2')
```
Comment (claudijd) [inline lib/tenableio_scan_handler.py]: Should our regions be configurable or is this just the default if a region isn't specified?

File: tests/test_ssh_observatory_scanner.py [hunk 5]
```diff
@@ -29,6 +29,8 @@ def test_setting_api_url(self):
         # Unstub env var for testing
         os.environ["SSHOBS_API_URL"] = ""
 
+    # NOTE: Temporary, as SSH Observatory API is broken
```
Comment (claudijd) [inline tests/test_ssh_observatory_scanner.py]: No longer broken, FYI

File: docs/usage.rst (hunk 1 above)
Comment (caggle) [reply docs/usage.rst]: Fixed.

File: lib/custom_exceptions.py (hunk 2 above)
Comment (caggle) [reply lib/custom_exceptions.py]: They are the custom exceptions I use to control the state machine, in order to perform conditional branching / retries. For instance, if a function raises `TenableScanRunningException` the state machine waits for a period of time before checking again.

See [this](https://github.com/mozilla/vautomator-serverless/blob/04355d127cb6c71b6518ae84a6c3fced0c8ed719/serverless.yml#L356-L369) as to how these are used.

File: lib/tenableio_scan_handler.py (hunk 3 above)
Comment (caggle) [reply lib/tenableio_scan_handler.py]: Yep, removed.

File: lib/tenableio_scan_handler.py (hunk 4 above)
Comment (caggle) [reply lib/tenableio_scan_handler.py]: I don't think it's the default, we are forcefully specifying it.
The reasons are:
- Everything about vautomator is currently deployed in `us-west-2`
- Some time ago, AWS gurus in the team asked people to use 

In saying this, it is not very configurable though, yeah. I will add an environment variable in `serverless.yml` for region, and use that in here to make it more configurable.

File: tests/test_ssh_observatory_scanner.py (hunk 5 above)
Comment (caggle) [reply tests/test_ssh_observatory_scanner.py]: Yep, thanks will uncomment.

**Pull request merged**